src/app/dashboard-widgets/create-work-item-widget/create-work-item-widget.component.ts

```diff
@@ -14,41 +14,73 @@ import { filterOutClosedItems, WorkItemsData } from '../../shared/workitem-utils
 export class CreateWorkItemWidgetComponent implements OnInit {
 
   @Input() userOwnsSpace: boolean;
+  @Input() loggedInUser: User;
+  @Input() currentSpace: Space;
 
-  private _myWorkItems: ConnectableObservable<WorkItem[]>;
-  myWorkItemsCount: Observable<number>;
+  private _myWorkItems: WorkItem[];
+  myWorkItemsCount: number;
   contextPath: Observable<string>;
+  subscriptions: Subscription[] = [];
 
   constructor(
     private filterService: FilterService,
     private workItemService: WorkItemService,
-    private userService: UserService,
     private contexts: Contexts
   ) {}
 
   ngOnInit() {
     this.contextPath = this.contexts.current.pipe(map(context => context.path));
-    this._myWorkItems = this.userService
-      .getUser().pipe(
-      tap(() => this.workItemService.buildUserIdMap()),
-      switchMap(() => this.userService.loggedInUser),
-      map(user => this.filterService.queryBuilder('assignee', this.filterService.equal_notation, user.id)),
-      switchMap(filters => this.workItemService
-        .getWorkItems(100000, {expression: filters})),
-      map((val: WorkItemsData) => val.workItems),
-      map(workItems => filterOutClosedItems(workItems)),
-      // Resolve the work item type
-      tap(workItems => workItems.forEach(workItem => this.workItemService.resolveType(workItem))),
-      tap(workItems => workItems.forEach(workItem => this.workItemService.resolveAreaForWorkItem(workItem))),
-      // MUST DO creator after area due to bug in planner
-      tap(workItems => workItems.forEach(workItem => this.workItemService.resolveCreator(workItem))),
-      publishReplay(1)) as ConnectableObservable<WorkItem[]>;
-    this._myWorkItems.connect();
-    this.myWorkItemsCount = this._myWorkItems.pipe(map(workItems => workItems.length));
+    this.getCurrentSpaceWorkItems();
   }
 
-  get myWorkItems(): Observable<WorkItem[]> {
+  ngOnChanges() {
+    this.getCurrentSpaceWorkItems();
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions.forEach(sub => {
+      sub.unsubscribe();
+    });
+  }
+
+  get myWorkItems(): WorkItem[] {
     return this._myWorkItems;
   }
 
+  getCurrentSpaceWorkItems() {
+    const assigneeQuery = this.filterService.queryJoiner(
+      {},
+      this.filterService.and_notation,
+      this.filterService.queryBuilder(
+        'assignee', this.filterService.equal_notation, this.loggedInUser.id
+      )
+    );
+    const spaceQuery = this.filterService.queryBuilder(
+      'space', this.filterService.equal_notation, this.currentSpace.id
+    );
+    const filters = this.filterService.queryJoiner(
+      assigneeQuery, this.filterService.and_notation, spaceQuery
+    );
+    this.subscriptions.push(
+      this.workItemService
+       .getWorkItems(100000, {expression: filters}).pipe(
+          map((val: WorkItemsData) => val.workItems),
+          map((workItems: WorkItem[]) => filterOutClosedItems(workItems)),
```

Comment:
Why are we not adding the filters while making the request for workitems?

Reply:
What do you mean by adding filters while making request? We are adding filters here. Which filters are you talking about?

Reply:
To filter out the closed work items. We can do that while making the request to the backend.
In this case, we are first fetching the workitems and then filter closed workitem in `filterOutClosedItems` method.

Reply:
That'd be great if this could be done in the request, in follow up PRs this could be ported to the other work-item widgets.

Reply:
@sahil143 Is there a documentation for query language where i can get info of how to build this query join it with present one?

Reply:
@sahil143 Adding a method in `workitems-util` won't work since it will need to use `FilterService` which is an `Injectable`. That is why it can only be used by injecting it in an Angular Component or Service. Will either need to create another service in `fabric8-ui` which will build and return filter queries or duplicate the same code everywhere the filter is needed.

Reply:
What we can do here is pass injected `FilterService` as a parameter to the method.

Reply:
That will solve the problem. But that doesn't seem like a good solution. Passing injectable service as a method parameter seems wrong to me because then the method will have to rely on the calling component to pass correct service to it which means the code can break if correct service is not passed to it.

Reply:
> But that doesn't seem like a good solution

Why?

> Passing injectable service as a method parameter seems wrong to me because then the method will have to rely on the calling component to pass correct service to it which means the code can break if correct service is not passed to it

You can do type checking. If the passed instance is not of the correct type. It will fail during compile time.
This will look something like this
```ts
closedWorkitemStateQuery(filterService: FilterService) {
....
}
```